src/azul/plugins/metadata/anvil/indexer/transform.py

```diff
@@ -410,7 +414,13 @@ def _file(self, file: EntityReference) -> MutableJSON:
                             uuid=file.entity_id)
 
     def _only_dataset(self) -> MutableJSON:
-        return self._dataset(one(self._entities_by_type['dataset']))
+        dataset = one(self._entities_by_type['dataset'])
+        try:
+            return self._dataset(dataset)
+        except KeyError:
+            entity_type = self.entity_type()
+            assert entity_type in {'bundles', 'datasets'}, entity_type
```

Comment:
Due to this assertion I have the feeling the handling of the special case can be pushed down into the bundle and dataset transformer or their common parent, singleton transformer. Also, why do we expect the KeyError for bundles? I am not a big fan of catching KeyError (or any exception) when the corresponding `[]` (or operation) isn't directly in the body of the try. 

Also, the only call-site to `_dataset` is here, so why even have two methods?

I recognized that this is a laundry list of objections, but this does smell a little bit.

Reply:
> Also, why do we expect the KeyError for bundles?

Because DUOS bundles now produce a `bundle` contribution in addition to a `dataset` contribution. Both contributions contain a "dataset" inner entity whose only fields are "description" and "document_id". The `KeyError` is due to the absence of the other dataset fields, e.g. "title".